BondageClub/Scripts/Typedef.d.ts

```diff
@@ -123,6 +123,73 @@ declare function LoseFocus(event: MouseEvent): void;
 
 //#endregion
 
+//#region Chat
+
+interface ChatRoom {
+	Name: string;
+	Description: string;
+	Admin: number[];
+	Ban: number[];
+	Limit: number;
+	Game: any;
+	Background: string;
+	Private: boolean;
+	Locked: boolean;
+	BlockCategory: "Photos";
+	Character?: Character[];
+}
+
+type StimulationAction = "Flash" | "Kneel" | "Walk" | "StruggleAction" | "StruggleFail";
+
+type MessageActionType = "Action" | "Chat" | "Whisper" | "Emote" | "Activity" | "Hidden" |
+ "LocalMessage" | "ServerMessage";
+
+type MessageContentType = string;
+
+interface ChatMessageDictionaryEntry {
+	[k: string]: any;
+	Tag?: string;
+	Text?: string;
+	MemberNumber?: number;
+}
+
+type ChatMessageDictionary = ChatMessageDictionaryEntry[];
+
+interface IChatRoomMessageBasic {
+	Content: MessageContentType;
+	Sender: number;
+	// SourceMemberNumber: number;
+}
+
+interface IChatRoomMessage extends IChatRoomMessageBasic {
+	Type: MessageActionType;
+	Dictionary?: ChatMessageDictionary;
+	Timeout?: number;
+}
+
+interface IChatRoomSyncBasic {
+	SourceMemberNumber: number
+}
+
+interface IChatRoomSyncMessage extends IChatRoomSyncBasic, ChatRoom {}
+
+//#endregion
+
+//#region FriendList
+
+interface IFriendListBeepLogMessage {
+	MemberNumber: number;
```

Comment:
`null` is valid, if from NPC
```suggestion
	MemberNumber: number | null;
```